torchvision/models/video/swin_transformer.py

```diff
@@ -0,0 +1,409 @@
+# Modified from 2d Swin Transformers in torchvision:
+# https://github.com/pytorch/vision/blob/main/torchvision/models/swin_transformer.py
+
+from functools import partial
+from typing import Callable, List, Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import nn, Tensor
+
+from ..swin_transformer import PatchMerging, SwinTransformerBlock
+
+__all__ = ["SwinTransformer3d"]
+
+
+def _compute_pad_size_3d(size_dhw: Tuple[int, int, int], patch_size: Tuple[int, int, int]) -> Tuple[int, int, int]:
+    pad_size = [(patch_size[i] - size_dhw[i] % patch_size[i]) % patch_size[i] for i in range(3)]
+    return pad_size[0], pad_size[1], pad_size[2]
+
+
+def _compute_attention_mask_3d(
```

Comment:
I think there is no such function in 2d case. This is just in 3d

Reply:
I think we can refactor the 2d implementation a bit, the 3d looks much better here :smile: